### DISCUSSION TITLE
install homebrew on console.cloud.google.com using sudo

### DISCUSSION THREAD

**Discussion by eliekh05** [Homebrew/discussions#3220] [Linux]
install homebrew on console.cloud.google.com using sudo error
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
==> Select a Homebrew installation directory:
- Enter your password to install to /home/linuxbrew/.linuxbrew (recommended)
- Press Control-D to install to /home/eliekh05/.linuxbrew
- Press Control-C to cancel installation

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

[sudo] password for eliekh05:
```
i can install it without using sudo but i want root access and this happens above

**Comment by SMillerDev** [comment]
None of this looks like an error, did you forget to copy this?

**Reply by eliekh05** [reply]
i did copy it

**Reply by SMillerDev** [reply]
So what is the part that seems like an error to you?

**Reply by eliekh05** [reply]
the problem is not running homebrew install script using sudo privileges on console.cloud.google.com cloud shell feature

**Reply by Bo98** [reply]
Using sudo shouldn't be an issue in Cloud Shell, you might need to check some settings.

If you however don't _want_ to use sudo then follow the prompts and where it says you can press Control-D, which will install into a place that doesn't need root privileges. Note that by doing so you cannot use all pre-built bottles (since some have baked in references to the default install directory), so some formulae will be installed from source instead, which can be slow and more prone to failing.

**Reply by eliekh05** [reply]
what settings should i change

**Reply by Bo98** [reply]
Looks like it's linked with `verifypw` setting.

But it also looks like you can bypass the Cloud Shell password issue by running non-interactively: `NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

This assumes default install settings, but if you've ever installed Homebrew before then it shouldn't be anything surprising:

```
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/linuxbrew/.linuxbrew/Homebrew
==> The following new directories will be created:
/home/linuxbrew/.linuxbrew/bin
/home/linuxbrew/.linuxbrew/etc
/home/linuxbrew/.linuxbrew/include
/home/linuxbrew/.linuxbrew/lib
/home/linuxbrew/.linuxbrew/sbin
/home/linuxbrew/.linuxbrew/share
/home/linuxbrew/.linuxbrew/var
/home/linuxbrew/.linuxbrew/opt
/home/linuxbrew/.linuxbrew/share/zsh
/home/linuxbrew/.linuxbrew/share/zsh/site-functions
/home/linuxbrew/.linuxbrew/var/homebrew
/home/linuxbrew/.linuxbrew/var/homebrew/linked
/home/linuxbrew/.linuxbrew/Cellar
/home/linuxbrew/.linuxbrew/Caskroom
/home/linuxbrew/.linuxbrew/Frameworks
```
